qiita_db/processing_job.py

```diff
@@ -0,0 +1,307 @@
+            return qdb.sql_connection.TRN.execute_fetchlast()
+
+    @classmethod
+    def create(cls, user, parameters):
+        """Creates a new job in the system
+
+        Parameters
+        ----------
+        user : qiita_db.user.User
+            The user executing the job
+        parameters : qiita_db.software.Parameters
+            The parameters of the job being executed
+
+        Returns
+        -------
+        qiita_db.processing_job.ProcessingJob
+            The newly created job
+        """
+        with qdb.sql_connection.TRN:
+            # Generate the job_id
+            job_id = str(uuid4())
+            while cls.exists(job_id):
+                job_id = str(uuid4())
+
+            sql = """INSERT INTO qiita.processing_job
+                        (processing_job_id, email, command_id,
+                         command_parameters_id, processing_job_status_id)
+                     VALUES (%s, %s, %s, %s, %s)"""
+            status = qdb.util.convert_to_id("queued", "processing_job_status")
+            sql_args = [job_id, user.id, parameters.command.id,
+                        parameters.id, status]
+            qdb.sql_connection.TRN.add(sql, sql_args)
+            return cls(job_id)
+
+    @property
+    def user(self):
+        """The user that launched the job
+
+        Returns
+        -------
+        qiita_db.user.User
+            The user that launched the job
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT email
+                     FROM qiita.processing_job
+                     WHERE processing_job_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            email = qdb.sql_connection.TRN.execute_fetchlast()
+            return qdb.user.User(email)
+
+    @property
+    def command(self):
+        """The command that the job executes
+
+        Returns
+        -------
+        qiita_db.software.Command
+            The command that the job executes
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT command_id
+                     FROM qiita.processing_job
+                     WHERE processing_job_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            cmd_id = qdb.sql_connection.TRN.execute_fetchlast()
+            return qdb.software.Command(cmd_id)
+
+    @property
+    def parameters(self):
+        """The parameters used in the job's command
+
+        Returns
+        -------
+        qiita_db.software.Parameters
+            The parameters used in the job's command
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT command_id, command_parameters_id
+                     FROM qiita.processing_job
+                     WHERE processing_job_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            res = qdb.sql_connection.TRN.execute_fetchindex()[0]
+            return qdb.software.Parameters(
```

Comment:
In order to instantiate the Parameters we need a commands object. The problem is that is not sufficient to know the parameter id, we also need to know to which command this parameter set belongs to.

i.e. Parameter id 1 can be from command 1, command 2, etc...

I think that my next PR will simplify this, as we will not have a table for each set of parameters as we have right now - I had it this way because this is how it would work with a table per command